### PR TITLE
feat(FR-3373): extend the terminology gate to user-manual prose

### LINKED ISSUE #8385: Extend the deterministic terminology gate to user-manual prose (docs CHECK 1), promote CHECK 3, and fix the resulting drift
## Problem

FR-3311's terminology gate is enforced on **one side only**. `scripts/check-terminology-i18n.mjs` hard-blocks (`--strict`, FR-3049) on the two i18n stores, but the **user manual prose** — the termbase's own home, `packages/backend.ai-webui-docs/src/{en,ko,ja,th}/**/*.md` — has no deterministic checker at all. The only docs-side terminology check today is the LLM `docs-lint` agent, which is non-deterministic and not wired into CI.

The asymmetry is not theoretical. With the same `terminology.json` that hard-gates i18n, a hand-rolled scan of the manual finds live drift the gate cannot see:

- `packages/backend.ai-webui-docs/src/ko/admin_menu/admin_menu.md:1319` and `:1342` use **슈퍼관리자**, which is an `avoid[]` row (`슈퍼관리자` → `슈퍼 관리자`). The other 26 occurrences across the ko manual are correct. i18n is clean; only the docs drifted.

`TERMINOLOGY.md`'s own Rename / Deprecation Checklist makes the gap explicit: step 3 says "Update the 4 doc languages — replace prose occurrences of the old term", but step 4's verification (`pnpm run lint:terminology`) only confirms _i18n labels_. Nothing verifies step 3.

## Secondary findings (same surface, same PR)

1. `en | organization → domain` is an imprecise blocking row. It is bare (`context: null`), so it is ERROR-severity. The manual uses "organization" 13 times in the ordinary English sense ("match the organization's hardware", "the organization's corporate identity") — none of them mean a Backend.AI _domain_. FR-3051 established the precision rule for non-English rows; the same rule is needed here. Left as-is, the docs gate cannot be blocking on day one, and any future i18n string containing "your organization" hard-blocks `verify.sh`.
1. **CHECK 3's promotion precondition is met.** The epic gates "CHECK 3 default-on" on clearing the FR-3302 candidate backlog. `node scripts/check-terminology-i18n.mjs --check3` now reports **0 candidates**. It can be flipped on by default (it is always report-only and never affects the exit code).
1. **Two governance docs went stale when FR-3049 flipped the gate to blocking:**
   - `packages/backend.ai-webui-docs/TERMINOLOGY.md` (New-Term Gate) still says the checker "is heuristic and warn-only — it never blocks a merge".
   - `.github/workflows/terminology-selftest.yml`'s header comment still says CHECK 1 is "wired report-only into scripts/verify.sh".
1. **FR-3311's open "not yet specified" item is already answered.** `I18N_GLOBS` in the checker already includes `packages/backend.ai-ui/src/locale`, so the blocking gate does cover the component-library store (the highest-blast-radius one).

## Scope

- Add a **docs prose source** to `scripts/check-terminology-i18n.mjs` (`--docs`), reusing the existing CHECK 1 matcher, severity model, code-context skipping, allowlist, and inline marker. Markdown-aware skips: fenced code blocks, inline code, link/image targets, HTML comments, front matter, reference-link definitions.
- Feed docs stores to **CHECK 1 only**. CHECK 2 (per-file segment near-duplicates) is meaningless for markdown, and CHECK 3 on docs prose would flood on Title-Case UI labels — both stay i18n-only.
- Fix the two genuine `슈퍼관리자` drifts in the ko manual.
- Context-qualify the `en | organization` avoid row so ordinary English usage does not hard-block, matching the FR-3051 precision rule.
- Promote CHECK 3 to default-on (report-only).
- Wire `--docs --strict` into `.github/workflows/docs-checks.yml`, which already triggers only on `packages/backend.ai-webui-docs/**`.
- Refresh the two stale governance statements, and add a docs-prose verification step to the Rename / Deprecation Checklist.

## Acceptance criteria

- `node scripts/check-terminology-i18n.mjs --docs --strict` exits 0 on a clean tree with **0 blocking findings** (clean baseline — the gate can block from day one).
- Reintroducing `슈퍼관리자` into a ko manual page is caught by the checker.
- `docs-checks` CI runs the docs terminology gate on any PR touching the manual.
- `bash scripts/verify.sh` ends with `=== ALL PASS ===`, and `terminology-selftest.yml`'s self-test still passes (the termbase is touched, so that workflow fires).

Parent map: FR-3311 (Terminology governance — enforce a blocking i18n terminology gate).

JIRA Issue: FR-3373

### PR DESCRIPTION
**Pull request by yomybaby**
Resolves #8385 (FR-3373)

## Why

The terminology gate is enforced on one side only. `check-terminology-i18n.mjs` hard-blocks (`--strict`, FR-3049) on the two i18n stores, but the **user manual** — the termbase's own home — had no deterministic checker at all. `TERMINOLOGY.md`'s Rename / Deprecation Checklist makes the hole explicit: step 4 verifies step 2 (UI locales) via `lint:terminology`, and nothing verifies step 3 ("update the 4 doc languages").

It was not theoretical. Running the same `avoid[]` rows over the manual found live drift the gate could not see:

```
warn  [packages/backend.ai-webui-docs/src/ko] admin_menu/admin_menu.md:1319
      term "슈퍼관리자" -> use "슈퍼 관리자"
warn  [packages/backend.ai-webui-docs/src/ko] admin_menu/admin_menu.md:1342
      term "슈퍼관리자" -> use "슈퍼 관리자"
```

The other 26 occurrences across the ko manual are correct. i18n was clean; only the docs drifted.

## What changed

**Docs prose source in the checker.** A line-oriented markdown reader feeds the existing CHECK 1 matcher — same termbase, same severity model, same allowlist and inline marker. Non-prose markdown (fenced code, inline code, link/image targets, HTML comments, front matter) is masked **index-preserving**, replacing each character with a space, so reported `spans` still point at the real column of the real line. The report prints the original line; the matcher sees the masked one. ON by default, `--no-docs` opts out.

**Docs see context-free avoid rows only.** A `context` qualifier means "this word is only wrong in one sense" — a judgement call. On short UI labels those rows are worth a warning; over 120 markdown files of long-form prose they are a permanent false-positive stream. Concretely, applying them produced 18 findings, every one legitimate English: "you can **group** three GPUs", "your **organization**'s SAML identity provider", the LGPL text in `license_agreement.md`. Judgement calls on prose stay with docs-lead / the `docs-lint` agent. CHECK 2 (per-file segment near-duplicates — markdown has no leaf-key segment) and CHECK 3 (Title-Case noun miner — the manual is written in exactly that register) stay i18n-only for the same signal-to-noise reason.

**Termbase precision fix.** `en | organization -> domain` was a *bare* row, so ERROR-severity and blocking, yet it fires on ordinary English. Any future i18n string containing "your organization" would have hard-blocked `verify.sh` on a perfectly correct label. It is now context-qualified (`as a Backend.AI scope`) — the same precision rule FR-3051 established for non-English rows, applied to the English side. `TERMINOLOGY.md` regenerated.

**CHECK 3 promoted to default-on.** The FR-3311 map gated this on clearing the FR-3302 candidate backlog; `--check3` now reports **0 candidates**. It remains always report-only and never affects the exit code.

It stays *off* in `verify.sh` (`--no-check3`), deliberately: `run_check` pipes every step through `tail -20`, and CHECK 3's always-present section costs ~5 of those lines — enough that on a failing run a real CHECK 1 finding would scroll off the top and the harness would report a failure without showing what failed. CHECK 3 is advisory, so it belongs in `pnpm run lint:terminology` rather than in the pass/fail harness.

**CI wiring.** `docs-checks.yml` runs the checker, and it already triggers only on `packages/backend.ai-webui-docs/**`. The checker is dependency-free (native Node), so it needs no extra install step.

**Stale governance statements.** Two documents still described the pre-FR-3049 world:
- `TERMINOLOGY.md`'s New-Term Gate said the checker "is heuristic and warn-only — it never blocks a merge". Rewritten into a per-row-kind table of what actually blocks.
- `terminology-selftest.yml`'s header said CHECK 1 is "wired report-only into scripts/verify.sh".

Rename-checklist step 4 now says `lint:terminology` verifies steps 2 **and** 3 together.

Also answers an FR-3311 "not yet specified" item: `I18N_GLOBS` already includes `packages/backend.ai-ui/src/locale`, so the blocking gate does cover the component-library store.

## Baseline

Clean on both sources, so the gate blocks from day one:

```
CHECK 1 — terminology drift (i18n VALUES + docs prose vs avoid[])
  no forbidden terms found in i18n values or manual prose.
  CHECK 1 summary: 0 blocking, 0 warn-only.
```

## Tests

`scripts/check-terminology-i18n.docs.test.ts` — 20 tests pinning the markdown reader (fences, fence-character matching, front matter only at the top, multi-line HTML comments, comments closing mid-line, the inline marker on the raw line, 1-based line keys, index-preserving masking) and the CHECK 1 rules over a docs store (context-free English blocks; context-qualified rows are skipped; the ko drift is caught at warn severity; an avoid row never crosses languages; a term inside code or a link target does not fire).

## Verification

```
bash scripts/verify.sh
=== ALL PASS ===

Terminology self-test (FR-3051): PASS — 350 assertion(s) hold.
pnpm exec vitest run scripts/ → 3 files, 66 tests passed
```

The self-test matters here because this PR touches `terminology.json`, which triggers `terminology-selftest.yml`.

## Follow-ups filed separately

- Elevating precision-gated non-English avoid rows from warn to error (they now ship with fixtures and a zero false-positive budget, so the original reason for blanket-warn has largely retired) — a policy call, not bundled here.
- The diff-aware CI merge-gate for i18n content that FR-3049 deferred.
- The CHECK 2 near-duplicate backlog (449 findings, mostly non-English casing/whitespace).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#8386** <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/8386?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/8386?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>flow:merge-queue</kbd> - adds this PR to the back of the merge queue
- <kbd>flow:hotfix</kbd> - for urgent changes, fast-track this PR to the front of the merge queue

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/lablup?ref=merge-queue-instructions-in-stack-comment&prId=8386).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by yomybaby** [comment]
All three findings were real and are fixed in 2483ee164.

**1. Multi-backtick spans** — confirmed by reproducing the old behaviour directly:

```
input:  Run ``scaling group ` here`` now.
old:    "Run   scaling group        ` now."   ← term leaked, would have produced a blocking finding
```

The regex backtracks its opening run down to one backtick and masks only the delimiters. Replaced with a hand-rolled scan implementing CommonMark's rule — a run of N backticks closes at the next run of exactly N — plus an unterminated-run case that emits the backticks as prose.

**2. Fence detection before HTML-comment state** — correct, and the consequence is worse than a missed line: the comment terminator gets swallowed and every finding in the rest of the document disappears. Comment closing now runs first, and fence detection reads the post-comment text.

**3. Contradicting default comment** — a leftover line from an incomplete edit. Removed.

Four tests added, one per defect plus two matcher edge cases (unterminated run; a short inner run must not close a longer span). `scripts/verify.sh` still ends `=== ALL PASS ===`.

File: scripts/check-terminology-i18n.mjs [hunk 1]
```diff
@@ -227,6 +264,141 @@ function listLocaleFiles(dir) {
...
+ * This is the markdown analogue of isInsideCodeContext() — same intent
+ * ("backticked tokens are not prose"), applied at line granularity because the
+ * non-prose regions here are markdown syntax rather than value-internal tokens.
+ * @param {string} line
+ */
+function maskMarkdownNonProse(line) {
+  return (
+    line
+      // Inline code spans: `x`, ``x with a ` in it``.
+      .replace(/(`+)[^`]*?\1/g, maskRun)
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply scripts/check-terminology-i18n.mjs]: Correct, and reproduced before fixing — the old regex backtracks its opening run down to a single backtick:

```
input:  Run ``scaling group ` here`` now.
old:    "Run   scaling group        ` now."
```

Only the delimiters were masked, so a code sample containing a deprecated term could have produced a *blocking* finding.

Fixed in 2483ee164: replaced with a hand-rolled scan implementing CommonMark's rule — a run of N backticks closes at the next run of exactly N — plus an unterminated-run branch that emits the backticks as prose. Three tests cover it: the multi-backtick span, the unterminated run, and "a short inner run must not close a longer span".

File: scripts/check-terminology-i18n.mjs [hunk 2]
```diff
@@ -1038,9 +1233,18 @@ function parseArgs(argv) {
     // --check2. Refining it into a high-signal check is tracked as a follow-up.
     runCheck2: false,
     // CHECK 3 (unknown capitalized user-facing noun) is OFF by default: it is a
-    // heuristic candidate-term suggester. Opt in with --check3. It is always
-    // report-only and never affects the exit code (mirrors CHECK 2's gating).
-    runCheck3: false,
+    // CHECK 3 (unknown capitalized user-facing noun) is ON by default since
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply scripts/check-terminology-i18n.mjs]: Right — leftover from an incomplete edit; the old opening line survived above the new one. Removed in 2483ee164, so `parseArgs` now carries a single accurate default comment for CHECK 3.

File: scripts/check-terminology-i18n.mjs [hunk 3]
```diff
@@ -227,6 +264,141 @@ function listLocaleFiles(dir) {
...
+      continue;
+    }
+    if (inFrontMatter) {
+      if (/^---\s*$/.test(line)) inFrontMatter = false;
+      continue;
+    }
+
+    // Fenced code blocks, closed by a fence of the same character.
+    const fence = line.match(/^\s*(`{3,}|~{3,})/);
+    if (fence) {
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply scripts/check-terminology-i18n.mjs:430]: Confirmed, and the blast radius is worse than a single missed line: the fence swallows the comment terminator, so `inFence` stays true and **every** finding in the rest of the document disappears silently.

Fixed in 2483ee164 — the `inHtmlComment` tail is consumed first, and fence detection then reads the post-comment `text` rather than the raw line. Regression test added ("does not let a fence inside an HTML comment swallow the rest of the file"): a commented-out ```bash block followed by prose, asserting the prose is still collected.
